### PR TITLE
Airlock Notifier: Provide a link to the request in the UI in the email

### LINKED ISSUE #2756: Add a link to the UI screen for Airlock Request to the notification email
## Description

As a TRE Researcher or Airlock Manager,
I'd like to see a link to UI screen for the Airlock request in my notification email
So that I can navigate to it right away

### PR DESCRIPTION
**Pull request by tanya-borisova**
# Resolves #2756 

## What is being addressed

Provide a link to the Airlock request in the notification email for the user to easily navigate to the request. 

## How is this addressed

- Pass in TRE url to the airlock notifier bundle 
- Change short_workspace_id to workspace_id to be able to construct a link to the Airlock request

### CODE REVIEW

File: templates/shared_services/airlock_notifier/template_schema.json [hunk 1]
```diff
@@ -25,14 +25,22 @@
...
       "type": "string",
       "title": "SMTP From Email",
       "description": "SMTP From Email",
       "updateable": false
+    },
+    "tre_url": {
+      "$id": "#/properties/tre_url",
+      "type": "string",
+      "title": "TRE URL",
+      "description": "If your TRE URL is different from ${TRE_ID}.${LOCATION}.cloudapp.azure.com, please enter it here",
```
Comment (tanya-borisova) [inline templates/shared_services/airlock_notifier/template_schema.json:42]: Maybe this message can be improved? In here I can't substitute TRE_ID and LOCATION with the actual values

Comment (damoodamoo) [reply templates/shared_services/airlock_notifier/template_schema.json:42]: I think it's ok - maybe add an example of the url? Although you sort of have one there already.

**Review by damoodamoo — APPROVED** [review]

**Review by martinpeck — APPROVED** [review]
Thanks @tanya-borisova

File: templates/shared_services/airlock_notifier/template_schema.json (hunk 1 above)
Comment (tamirkamara) [reply templates/shared_services/airlock_notifier/template_schema.json:42]: Should this also include a validation expression?

File: api_app/event_grid/event_sender.py [hunk 2]
```diff
@@ -28,14 +28,14 @@ async def send_status_changed_event(airlock_request: AirlockRequest, previous_st
...
     request_id = airlock_request.id
     status = airlock_request.status.value
-    short_workspace_id = airlock_request.workspaceId[-4:]
+    workspace_id = airlock_request.workspaceId
     snake_case_emails = {re.sub(r'(?<!^)(?=[A-Z])', '_', role_name).lower(): role_id for role_name, role_id in emails.items()}
 
     airlock_notification = EventGridEvent(
         event_type="airlockNotification",
-        data=AirlockNotificationData(request_id=request_id, event_type="status_changed", event_value=status, emails=snake_case_emails, workspace_id=short_workspace_id).__dict__,
+        data=AirlockNotificationData(request_id=request_id, event_type="status_changed", event_value=status, emails=snake_case_emails, workspace_id=workspace_id).__dict__,
```
Comment (tamirkamara) [inline api_app/event_grid/event_sender.py:36]: Just for my info - what have we used this value before this change?

Comment (tanya-borisova) [reply api_app/event_grid/event_sender.py:36]: There was a short workspace URL, which only included the  last 4 characters of the workspace.  (Same is true for StatusChanged event).

I'm not sure why, though! Do you know? To me it makes more sense to send a full workspace ID in the event as the consumer can always take the last 4 characters of it.

File: templates/shared_services/airlock_notifier/template_schema.json (hunk 1 above)
Comment (tanya-borisova) [reply templates/shared_services/airlock_notifier/template_schema.json:42]: Added one

**Pull request merged**